### ISSUE TITLE
Predis throwing a 'ServerException'

### ISSUE THREAD

**Issue by davidruigrok** [laravel/horizon#118]
After installing Horizon (server is running through Forge), Predis throws a ServerException when dispatching a new job.
`[Predis\Response\ServerException] ERR value is not a valid float`
Created a 'empty' job to test, but also throws the same exception.
Dispatching test job `dispatch((new \App\Jobs\Test)->onQueue('default'));` 
Horizon dashboard is working, configuration is the same as the one from the horizon documentation.
When changing queue driver to 'database' everything works fine.

**Comment by themsaid** [comment]
Can't replicate your issue, seems to be working fine to me.

**Comment by davidruigrok** [comment]
It only seems to throw the error when dispatching from an artisan command:
`Artisan::command('inspire', function () {
    dispatch((new \App\Jobs\Test)->onQueue('default'));
});`
When dispatching from an action on a controller it works fine.

Trace when fails:
`[Predis\Response\ServerException]
ERR value is not a valid float     
Exception trace:
 () at /home/forge/site/vendor/predis/predis/src/Pipeline/Pipeline.php:102
 Predis\Pipeline\Pipeline->exception() at /home/forge/site/vendor/predis/predis/src/Pipeline/Pipeline.php:146
 Predis\Pipeline\Pipeline->executePipeline() at /home/forge/site/vendor/predis/predis/src/Pipeline/Pipeline.php:165
 Predis\Pipeline\Pipeline->flushPipeline() at /home/forge/site/vendor/predis/predis/src/Pipeline/Pipeline.php:214
 Predis\Pipeline\Pipeline->execute() at /home/forge/site/vendor/predis/predis/src/Client.php:445
 Predis\Client->createPipeline() at /home/forge/site/vendor/predis/predis/src/Client.php:396
 Predis\Client->sharedContextFactory() at /home/forge/site/vendor/predis/predis/src/Client.php:418
 Predis\Client->pipeline() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Redis/Connections/Connection.php:96
 Illuminate\Redis\Connections\Connection->command() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Redis/Connections/Connection.php:108
 Illuminate\Redis\Connections\Connection->__call() at /home/forge/site/vendor/laravel/horizon/src/Repositories/RedisJobRepository.php:221
 Laravel\Horizon\Repositories\RedisJobRepository->pushed() at /home/forge/site/vendor/laravel/horizon/src/Listeners/StoreJob.php:37
 Laravel\Horizon\Listeners\StoreJob->handle() at n/a:n/a
 call_user_func_array() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php:368
 Illuminate\Events\Dispatcher->Illuminate\Events\{closure}() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php:200
 Illuminate\Events\Dispatcher->dispatch() at /home/forge/site/vendor/laravel/horizon/src/RedisQueue.php:155
 Laravel\Horizon\RedisQueue->event() at /home/forge/site/vendor/laravel/horizon/src/RedisQueue.php:62
 Laravel\Horizon\RedisQueue->Laravel\Horizon\{closure}() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Support/helpers.php:1035
 tap() at /home/forge/site/vendor/laravel/horizon/src/RedisQueue.php:63
 Laravel\Horizon\RedisQueue->pushRaw() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Queue/RedisQueue.php:82
 Illuminate\Queue\RedisQueue->push() at /home/forge/site/vendor/laravel/horizon/src/RedisQueue.php:46
 Laravel\Horizon\RedisQueue->push() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Queue/Queue.php:44
 Illuminate\Queue\Queue->pushOn() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php:177
 Illuminate\Bus\Dispatcher->pushCommandToQueue() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php:159
 Illuminate\Bus\Dispatcher->dispatchToQueue() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php:73
 Illuminate\Bus\Dispatcher->dispatch() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Foundation/Bus/PendingDispatch.php:86
 Illuminate\Foundation\Bus\PendingDispatch->__destruct() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php:382
 dispatch() at /home/forge/site/routes/console.php:6
 Illuminate\Foundation\Console\ClosureCommand->{closure}() at n/a:n/a
 call_user_func_array() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:29
 Illuminate\Container\BoundMethod::Illuminate\Container\{closure}() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:75
 Illuminate\Container\BoundMethod::callBoundMethod() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:31
 Illuminate\Container\BoundMethod::call() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Container/Container.php:549
 Illuminate\Container\Container->call() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Foundation/Console/ClosureCommand.php:55
 Illuminate\Foundation\Console\ClosureCommand->execute() at /home/forge/site/vendor/symfony/console/Command/Command.php:264
 Symfony\Component\Console\Command\Command->run() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Console/Command.php:167
 Illuminate\Console\Command->run() at /home/forge/site/vendor/symfony/console/Application.php:888
 Symfony\Component\Console\Application->doRunCommand() at /home/forge/site/vendor/symfony/console/Application.php:224
 Symfony\Component\Console\Application->doRun() at /home/forge/site/vendor/symfony/console/Application.php:125
 Symfony\Component\Console\Application->run() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Console/Application.php:82
 Illuminate\Console\Application->run() at /home/forge/site/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php:119
 Illuminate\Foundation\Console\Kernel->handle() at /home/forge/site/artisan:35`

**Comment by davidruigrok** [comment]
Somehow the problem only occurs when certain artisan commands are 'activated' in `App\Console\Kernel`.
When those commands are removed / commented it all works as intended. I will try to debug and find out what is causing the problem

**Comment by dennisoderwald** [comment]
+1

**Comment by themsaid** [comment]
@dennisoderwald would appreciate if you give some details, having different persons being able to replicate the issue helps us understand it.

**Comment by themsaid** [comment]
@davidruigrok how can I replicate this?

**Comment by davidruigrok** [comment]
@themsaid Try the following:
- Create a new console command with following __construct
```php
public function __construct() { 
    parent::__construct(); 
    setlocale(LC_ALL, 'nl_NL.utf8'); 
}
```
- Register command within `$commands` on `App\Console\Kernel`
- Create _empty_ test job (empty handle method is fine)
- Create test command which tries to dispatch a job
```php
Artisan::command('test', function () { 
    dispatch(new \App\Jobs\Test); 
});
```
- Try to execute _test_ command
`php artisan test -v`
- Throws:
```
[Predis\Response\ServerException] 
ERR value is not a valid float
```

**Comment by themsaid** [comment]
Still can't replicate, no errors.

**Comment by davidruigrok** [comment]
Then maybe it is triggered by a package or an installed extension. I will further debug to make sure horizon can be excluded for triggering this exception

**Comment by themsaid** [comment]
Check https://github.com/laravel/horizon/pull/123

**Comment by dennisoderwald** [comment]
@themsaid New error ```ERR Error running script (call to f_0ac1d0b183ba9e097017cf4800a6a65400ae3539): @user_script:11: user_script:11: attempt to perform arithmetic on a nil value```, same exception `Predis\Response\ServerException`.

**Comment by davidruigrok** [comment]
@themsaid Thanks, PR #123 fixed my issue, no ServerException anymore.

**Comment by dennisoderwald** [comment]
As I said before, I get a different error message, attached is a stacktrace:

```Predis\Response\ServerException ERR Error running script (call to f_0ac1d0b183ba9e097017cf4800a6a65400ae3539): @user_script:11: user_script:11: attempt to perform arithmetic on a nil value  
    vendor/predis/predis/src/Client.php:370 Predis\Client::onErrorResponse
    vendor/predis/predis/src/Client.php:335 Predis\Client::executeCommand
    vendor/predis/predis/src/Client.php:314 Predis\Client::__call
    vendor/laravel/framework/src/Illuminate/Redis/Connections/Connection.php:96 Illuminate\Redis\Connections\Connection::command
    vendor/laravel/framework/src/Illuminate/Redis/Connections/Connection.php:108 Illuminate\Redis\Connections\Connection::__call
    vendor/laravel/horizon/src/Repositories/RedisMetricsRepository.php:200 Laravel\Horizon\Repositories\RedisMetricsRepository::incrementQueue
    vendor/laravel/horizon/src/Listeners/UpdateJobMetrics.php:53 Laravel\Horizon\Listeners\UpdateJobMetrics::handle
    [internal] call_user_func_array
    vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php:368 Illuminate\Events\Dispatcher::Illuminate\Events\{closure}
    vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php:200 Illuminate\Events\Dispatcher::dispatch
    vendor/laravel/horizon/src/RedisQueue.php:155 Laravel\Horizon\RedisQueue::event
    vendor/laravel/horizon/src/RedisQueue.php:124 Laravel\Horizon\RedisQueue::deleteReserved
    vendor/laravel/framework/src/Illuminate/Queue/Jobs/RedisJob.php:84 Illuminate\Queue\Jobs\RedisJob::delete
    vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php:57 Illuminate\Queue\CallQueuedHandler::call
    vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php:76 Illuminate\Queue\Jobs\Job::fire
    vendor/laravel/framework/src/Illuminate/Queue/Worker.php:320 Illuminate\Queue\Worker::process
    vendor/laravel/framework/src/Illuminate/Queue/Worker.php:270 Illuminate\Queue\Worker::runJob
    vendor/laravel/framework/src/Illuminate/Queue/Worker.php:114 Illuminate\Queue\Worker::daemon
    vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php:101 Illuminate\Queue\Console\WorkCommand::runWorker
    vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php:85 Illuminate\Queue\Console\WorkCommand::handle
    [internal] call_user_func_array
    vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:29 Illuminate\Container\BoundMethod::Illuminate\Container\{closure}
    vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:87 Illuminate\Container\BoundMethod::callBoundMethod
    vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:31 Illuminate\Container\BoundMethod::call
    vendor/laravel/framework/src/Illuminate/Container/Container.php:549 Illuminate\Container\Container::call
    vendor/laravel/framework/src/Illuminate/Console/Command.php:180 Illuminate\Console\Command::execute
    vendor/symfony/console/Command/Command.php:264 Symfony\Component\Console\Command\Command::run
    vendor/laravel/framework/src/Illuminate/Console/Command.php:167 Illuminate\Console\Command::run
    vendor/symfony/console/Application.php:888 Symfony\Component\Console\Application::doRunCommand
    vendor/symfony/console/Application.php:224 Symfony\Component\Console\Application::doRun
    vendor/symfony/console/Application.php:125 Symfony\Component\Console\Application::run
    vendor/laravel/framework/src/Illuminate/Console/Application.php:88 Illuminate\Console\Application::run
    vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php:121 Illuminate\Foundation\Console\Kernel::handle
    artisan:37 [main]```

**Comment by themsaid** [comment]
@dennisoderwald you'll need to debug it further since the error is very vague.

**Comment by dennisoderwald** [comment]
@themsaid According to my `Log::debug()` commands the error occurs when all functions have already been executed. Wouldn't know how to continue debugging now. You got any ideas?

**Comment by themsaid** [comment]
clear your redis server and start over, maybe some corrupted records from old failures.

**Comment by dennisoderwald** [comment]
@themsaid I rebooted Redis and truncated all the databases. The error still occurs. Do you have another idea?

**Comment by dennisoderwald** [comment]
@themsaid It seems to be fixed with version v1.0.0 to v1.0.3.
